### ISSUE TITLE
DuckDuckGo Search Functionality Not Working with Next.js

### ISSUE THREAD

**Issue by klei30** [run-llama/create-llama#427]
I'm implementing DuckDuckGo search functionality in my Next.js application using `llamaindex`. However, the search feature is not working as expected. Also in the other apps that i have already deployed online the DuckDuckGo  search stop working.

 Below are the details of my implementation and the issues I'm encountering.

### **Steps to Reproduce**

npx create-llama@latest --pro
√ What is your project named? ... newinsuranceagent
√ Which template would you like to use? » Agentic RAG (e.g. chat with docs)
√ Which framework would you like to use? » NextJS
√ Would you like to set up observability? » Traceloop
√ Please provide your OpenAI API key (leave blank to skip): ...
√ Would you like to use a vector database? » Pinecone
√ Which data source would you like to use? » Use an example PDF
√ Would you like to add another data source? » No
√ Would you like to use LlamaParse (improved parser for RAG - requires API key)? ... No / Yes
√ Please provide your LlamaCloud API key (leave blank to skip): ...
√ Would you like to build an agent using tools? If so, select the tools here, otherwise just press enter » DuckDuckGo Search

### **Actual Behavior**

"user: search  in internet about last news in BBC 

AI: It seems there was an issue with the search. Let me try again to find information about BBC news"

and it keeps loading  without showing any other result

**Comment by leehuwuj** [comment]
@klei30 Yes, I think it's the issue of the DuckDuckGo scrape library. Could you try updating it to the latest version and see if that works?

**Comment by marcusschiesser** [comment]
@KleiAliaj mentioned in #433 that it's also not working for Python

**Comment by leehuwuj** [comment]
Hi @klei30 and @KleiAliaj, we just fixed the issue with the DuckDuckGo tool for the Python template. Unfortunately, the TypeScript version needs more effort because the duck-duck-scrape library hasn't been updated for a year. Please try using another search engine tool if you are using Typescript template in the meantime.

**Comment by marcusschiesser** [comment]
yes, the library is not working at the moment, see https://github.com/Snazzah/duck-duck-scrape/issues/140 - so we will:
1. remove duckduckgo from agentic rag template
2. generally deactivate the tool for TS

**Comment by marcusschiesser** [comment]
@klei30 currently you can use the python backend as a workaround. I couldn't find another TS library using duckduckgo that's not outdated. Want to give https://www.npmjs.com/package/@serptech/api a try?
